node-ui/src/pages/Applications.tsx

```diff
@@ -114,74 +123,127 @@ export default function ApplicationsPage() {
+                version: app.version,
+                source: app.source,
+                blob: app.blob,
+                contract_app_id: null,
+                name: 'local app',
+                description: null,
+                repository: null,
+                owner: null,
+              };
+            } else {
+              const [packageData, releaseData]: [
+                Package | null,
+                Release | null,
+              ] = await Promise.all([
+                getPackage(appMetadata.contractAppId),
+                getLatestRelease(appMetadata.contractAppId),
+              ]);
+
+              if (packageData) {
                 application = {
-                  id: app.id,
-                  version: app.version,
-                  source: app.source,
-                  blob: app.blob,
-                  contract_app_id: null,
-                  name: 'local app',
-                  description: null,
-                  repository: null,
-                  owner: null,
+                  ...app,
+                  contract_app_id: appMetadata.contractAppId,
+                  name: packageData?.name ?? '',
+                  description: packageData?.description,
+                  repository: packageData?.repository,
+                  owner: packageData?.owner,
+                  version: releaseData?.version ?? '',
                 };
-              } else {
-                const packageData: Package | null = await getPackage(
-                  appMetadata.contractAppId,
-                );
-
-                if (packageData) {
-                  application = {
-                    ...app,
-                    contract_app_id: appMetadata.contractAppId,
-                    name: packageData?.name ?? '',
-                    description: packageData?.description,
-                    repository: packageData?.repository,
-                    owner: packageData?.owner,
-                  };
-                }
               }
+            }
 
-              return application;
-            },
-          ),
-        );
-        var installed: Application[] = tempApplications.filter(
-          (app): app is Application => app !== null,
-        );
+            return application;
+          },
+        ),
+      );
+      var installed: Application[] = tempApplications.filter(
+        (app): app is Application => app !== null,
+      );
 
-        setApplications((prevState: Applications) => ({
-          ...prevState,
-          installed,
-        }));
-      }
-    };
+      setApplications((prevState: Applications) => ({
+        ...prevState,
+        installed,
+      }));
+    }
+  };
 
+  useEffect(() => {
     setApps();
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const uninstallApplication = async () => {
+    const contextList = (
+      await apiClient(showServerDownPopup).node().getContexts()
+    ).data?.contexts;
+    if (contextList?.length !== 0) {
```

Comment:
is it possible that this request returns error?

Reply:
Updated to handle error also